src/platform/X32/startup/soc.c

```
@@ -0,0 +1,157 @@
+
+BOOT_CODE static void relocate_runtime_image(void)
+{
+    volatile uint32_t *src = &__image_copy_load_start__;
+    volatile uint32_t *dst = &__image_copy_start__;
+    uint32_t *end = &__image_copy_end__;
+
+    if (dst != src)
+    {
+        boot_copy_words(dst, src, end);
+    }
+    dst[15] = src[250];
```

Cppcheck (2.20.0)

[error] The address of variable '__image_copy_start__' is accessed at non-zero index.

(objectIndex)